docs/proposals/annotations-downward-api.md

```diff
@@ -0,0 +1,60 @@
+# Exposing annotations via environment downward API
+
+Author: Michal Rostecki \<mrostecki@mirantis.com\>
+
+## Introduction
+
+Annotations of the pod can be taken through the Kubernetes API, but currently
+there is no way to pass them to the application inside the container. This means
+that annotations can be used by the core Kubernetes services and the user outside
+of the Kubernetes cluster.
+
+Of course using Kubernetes API from the application running inside the container
+managed by Kubernetes is technically possible, but that's an idea which denies
+the principles of microservices architecture.
+
+The purpose of the proposal is to allow to pass the annotation as the environment
+variable to the container.
+
+### Use-case
+
+The primary usecase for this proposal are StatefulSets. There is an idea to expose
+StatefulSet index to the applications running inside the pods managed by StatefulSet.
+Since StatefulSet creates pods as the API objects, passing this index as an
+annotation seems to be a valid way to do this. However, to finally pass this
+information to the containerized application, we need to pass this annotation.
+That's why the downward API for annotations is needed here.
+
+## API
+
+The exact `fieldPath` to the annotation will look like:
+
+```
+metadata.annotations['annotationKey']
```

Comment:
Not just talking about annotations.  Any key applies.  I don't think JSONPath demands single-quotes, but all examples I see are canonically single quote.